### ISSUE TITLE
[Suggestion - New Backend] Add OpenAI Scorer

### ISSUE THREAD

**Issue by julien-blanchon** [leibniz-csi/mm_grpo#29]
Hello,

I'm having a look at your codebase it's pretty good !

I would like to suggest you to add a vanilla OpenAIClient Scorer typically taking:
- A base url, either a local private one or a public one (OpenRouter, OpenAI, Local VLLM OpenAI endpoint, Local LM Studio ...).
- An optional api key.
- An optional pydantic schema to eventually use structured generation (usefull to ensure score structure)
- An optional factor K and mean method ("mean", "median", "trimmed_mean"). To reduce the reward score variance by generating multiple score and meaning the results with a given method. (PS: Another idea would be to directly use the token logits but this not possible for most API endpoints).

I'm using openrouter gemini-flash with DiffusionNFT to train on arbirary rewards, typically taking an evaluation prompt as well as the generated image. And it's working incredibly well. It's also super cost efficient and as it's literally taking few seconds to collect hundred of rewards. I'm also the same framework for video (also with Gemini).
